core/component.go

```diff
@@ -70,9 +70,12 @@ func (lhs *Component) Equal(rhs *Component) bool {
 	if lhs == nil || rhs == nil {
 		return false
 	}
+	// TODO: Name cannot be added as the name of the component will not match
```

Comment:
@xadhatter Flagging this code as it was something I found. When an event is returning back from httpsrv to the broker this check was failing. This is since the httpsrv doesn't have a name that matches the name of the adapter the request was for. Not sure if you'd like any changes here (I'll remove the code  if no changes).

Reply:
Def shouldn't be fixed here. It'll break other component checks. Generally I think the right approach is to make the adapter fit into the existing way components are represented. So if `Name` is not getting set properly figure out why and correct it there.